force-app/main/default/classes/SOQL.cls

```diff
@@ -312,50 +385,89 @@ public with sharing class SOQL {
      * @param Original Opportunity Id
      * @param List of cloned Opportunity Ids
      */
-    public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
+      public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
         // Query the original Opportunity Line Items
-
+        List<OpportunityLineItem> originalLineItems = [SELECT PricebookEntryId, Quantity, UnitPrice, Discount, Description FROM OpportunityLineItem WHERE OpportunityId = :origOppId];
+    
         // Create a list to hold the new line items
-
-        // Generate a random number between 1 and 3
-
+        List<OpportunityLineItem> newLineItems = new List<OpportunityLineItem>();
+    
+        Integer min = 1;
+        Integer max = 3;
+    
         // Loop through each cloned Opportunity Id
-
+        for (Id cloneOppId : cloneOppIds) {
             // Loop through each original line item
-
+            for (OpportunityLineItem origLineItem : originalLineItems) {
                 // Clone the original line item and add the random number to the quantity
-
+                OpportunityLineItem newLineItem = new OpportunityLineItem(
+                    OpportunityId = cloneOppId,
+                    PricebookEntryId = origLineItem.PricebookEntryId,
+                    UnitPrice = origLineItem.UnitPrice,
+                    Discount = origLineItem.Discount,
+                    Description = origLineItem.Description
+                );
+                
+                // Generate a random number between 1 and 3                
+                Double randomQuantityIncrease = Math.floor(Math.random() * (max - min + 1)) + min;   
```

Comment:
I would move this outside the for loop so its not recalculating each iteration. All of the opps can have the same random quantity